src/plugins/intel_cpu/src/plugin.cpp

```diff
@@ -171,260 +172,307 @@ Engine::~Engine() {
     executorManager()->clear("CPUCallbackExecutor");
 }
 
-static void TransformationUpToCPUSpecificOpSet(std::shared_ptr<ngraph::Function> nGraphFunc, const bool _enableLPT,
-                                               const bool _enableSnippets, const bool isLegacyApi) {
-    ngraph::pass::Manager manager;
-    manager.set_per_pass_validation(false);
-    manager.register_pass<ngraph::pass::InitNodeInfo>();
-
-    const bool useLpt =
-            _enableLPT &&
-        ngraph::pass::low_precision::LowPrecision::isFunctionQuantized(nGraphFunc);
-    auto defaultPrecisions = useLpt ? ngraph::pass::low_precision::precision_set::int8_support : std::vector<ov::element::Type>{};
-    bool hasINT16orINT32Levels = false;
-    if (useLpt) {
-        hasINT16orINT32Levels = ngraph::pass::low_precision::LowPrecision::isFQLevelsPresent(
+class Transformations {
+public:
+    Transformations(const std::shared_ptr<ngraph::Function>& ngraphFunction
+                    CPU_DEBUG_CAP_ENABLE(, const Config& config))
+        : nGraphFunc(ngraphFunction) CPU_DEBUG_CAP_ENABLE(, config(config)) {}
+
+    Transformations(const std::shared_ptr<ngraph::Function>& ngraphFunction,
+                    const bool enableLpt, const bool enableSnippets, const bool isLegacyApi
+                    CPU_DEBUG_CAP_ENABLE(, const Config& config))
+        : Transformations(ngraphFunction CPU_DEBUG_CAP_ENABLE(, config)) {
+        upToCpuSpecificOpSet(enableLpt, enableSnippets, isLegacyApi);
+        cpuSpecificOpSet();
+    }
+
+    void upToCpuSpecificOpSet(const bool enableLpt, const bool enableSnippets, const bool isLegacyApi) {
+        const bool useLpt = enableLpt &&
+                            ngraph::pass::low_precision::LowPrecision::isFunctionQuantized(nGraphFunc);
+
+        upToCpuSpecificOpSet_common(useLpt, isLegacyApi);
+
+        if (!useLpt && enableSnippets)
+            upToCpuSpecificOpSet_snippets();
+    }
+
+    void cpuSpecificOpSet(void) {
+        CPU_DEBUG_CAP_TRANSFORMATION_RETURN_OR_DUMP(Specific);
+
+        ConvertToCPUSpecificOpset(nGraphFunc);
+    }
+
+private:
+    const std::shared_ptr<ngraph::Function>& nGraphFunc;
+    using const_node_ptr = const std::shared_ptr<const ngraph::Node>;
+    CPU_DEBUG_CAP_ENABLE(const Config& config);
+
+    void upToCpuSpecificOpSet_common(const bool useLpt, const bool isLegacyApi) {
```

Comment:
As it was discussed by common transformations we mean ngraph::pass::CommonOptimizations>.
What about the following order:
1) common (CommonOptimizations)
2) cpu essentials or preLpt
3) lpt
4) postLpt
4) snippets

I think there is no need to combine them, we can just call them one after another.

Reply:
Discussed.
Let's split stages "naturally", with separate steps

Reply:
@EgorDuplensky, is this decision for both disabling and serialization?
that looks good to have the same logic/code for these features.

Reply:
leaving this open for final review